diff -urN gcl-2.4.0/bin/makefile gcl-2.4.0-new/bin/makefile
--- gcl-2.4.0/bin/makefile	Mon Dec  3 22:53:05 2001
+++ gcl-2.4.0-new/bin/makefile	Mon Dec  3 23:10:59 2001
@@ -1,5 +1,5 @@
 DEFS	= -DUNIX
-CFLAGS  = $(DEFS) -I../h
+CFLAGS  = $(DEFS) -I../h $(OPTFLAGS)
 CC=cc
 APPEND=../bin/append
 
diff -urN gcl-2.4.0/cmpnew/makefile gcl-2.4.0-new/cmpnew/makefile
--- gcl-2.4.0/cmpnew/makefile	Mon Dec  3 22:53:05 2001
+++ gcl-2.4.0-new/cmpnew/makefile	Mon Dec  3 23:09:53 2001
@@ -14,7 +14,7 @@
 
 COMPILE_FILE=$(PORTDIR)/saved_gcl $(PORTDIR) -system-p -c-file -data-file \
 	-o-file nil -h-file -compile
-CFLAGS	= -c -O -I../h 
+CFLAGS	= -c -O -I../h $(OPTFLAGS)
 
 .lsp.c:
 	@ ../xbin/if-exists $(PORTDIR)/saved_gcl \
diff -urN gcl-2.4.0/dos/makefile gcl-2.4.0-new/dos/makefile
--- gcl-2.4.0/dos/makefile	Mon Dec  3 22:53:05 2001
+++ gcl-2.4.0-new/dos/makefile	Mon Dec  3 23:10:45 2001
@@ -115,7 +115,7 @@
 
 DOS_ODIR=.
 
-CFLAGS	= -I. -I$(HDIR) $(ODIR_DEBUG)
+CFLAGS	= -I. -I$(HDIR) $(ODIR_DEBUG) $(OPTFLAGS)
 
 .s.o:
 	$(CC) -c $(OFLAG) $(CFLAGS) $*.c
diff -urN gcl-2.4.0/gcl-tk/makefile gcl-2.4.0-new/gcl-tk/makefile
--- gcl-2.4.0/gcl-tk/makefile	Mon Dec  3 22:53:05 2001
+++ gcl-2.4.0-new/gcl-tk/makefile	Mon Dec  3 23:10:05 2001
@@ -123,7 +123,7 @@
 # end makedefs
 
 
-CFLAGS1=$(CFLAGS) -I../o -I../h ${TK_INCLUDE} ${TCL_INCLUDE} ${TK_XINCLUDES}
+CFLAGS1=$(CFLAGS) -I../o -I../h ${TK_INCLUDE} ${TCL_INCLUDE} ${TK_XINCLUDES} $(OPTFLAGS) 
 
 
 all: gcltksrv   tkl.o tinfo.o  demos/gc-monitor.o gcltkaux
diff -urN gcl-2.4.0/go/makefile gcl-2.4.0-new/go/makefile
--- gcl-2.4.0/go/makefile	Mon Dec  3 22:53:05 2001
+++ gcl-2.4.0-new/go/makefile	Mon Dec  3 23:12:08 2001
@@ -18,14 +18,14 @@
 
 AUX_INFO=-aux-info $*.X
 .c.o:
-	$(CC) -c $(OFLAG) $(CFLAGS) $*.c $(AUX_INFO)
+	$(CC) -c $(OFLAG) $(CFLAGS) $(OPTFLAGS) $*.c $(AUX_INFO)
 .d.o:
 	$(DPP) $*
-	$(CC) -c $(OFLAG) $(CFLAGS) $*.c $(AUX_INFO)
+	$(CC) -c $(OFLAG) $(CFLAGS) $(OPTFLAGS) $*.c $(AUX_INFO)
 	rm $*.c
 
 .c.ini:
-	$(CC) -DNO_DEFUN $(CFLAGS) -E $*.c | grab_defs > $*.ini
+	$(CC) -DNO_DEFUN $(CFLAGS) $(OPTFLAGS) -E $*.c | grab_defs > $*.ini
 
 .d.ini:
 	$(DPP) $*
diff -urN gcl-2.4.0/lsp/makefile gcl-2.4.0-new/lsp/makefile
--- gcl-2.4.0/lsp/makefile	Mon Dec  3 22:53:05 2001
+++ gcl-2.4.0-new/lsp/makefile	Mon Dec  3 23:10:38 2001
@@ -16,7 +16,7 @@
 
 COMPILE_FILE=$(PORTDIR)/saved_gcl $(PORTDIR) -system-p -c-file -data-file \
 	-o-file nil -h-file -compile
-CFLAGS	= -c -O -I../h 
+CFLAGS	= -c -O -I../h $(OPTFLAGS)
 
 .lsp.c: 
 	@ ../xbin/if-exists $(PORTDIR)/saved_gcl \
diff -urN gcl-2.4.0/mp/makefile gcl-2.4.0-new/mp/makefile
--- gcl-2.4.0/mp/makefile	Mon Dec  3 22:53:05 2001
+++ gcl-2.4.0-new/mp/makefile	Mon Dec  3 23:11:29 2001
@@ -133,7 +133,7 @@
 	$(AS) $*.s -o $*.o
 
 .c.o:
-	$(CC)  -c $(OFLAG) -I../h -I. $(CFLAGS) $(ODIR_DEBUG) $*.c
+	$(CC)  -c $(OFLAG) -I../h -I. $(CFLAGS) $(ODIR_DEBUG) $(OPTFLAGS) $*.c
 
 mpi-386_no_under.o: mpi-386_no_under.s
 	gcc -traditional -c $*.s -o $*.o
diff -urN gcl-2.4.0/o/makefile gcl-2.4.0-new/o/makefile
--- gcl-2.4.0/o/makefile	Mon Dec  3 22:53:05 2001
+++ gcl-2.4.0-new/o/makefile	Mon Dec  3 23:13:18 2001
@@ -144,7 +144,7 @@
 
 # end makedefs
 X11_INCLUDE=${TK_XINCLUDES} ${HAVE_X11}
-CFLAGS	=  -I../gcl-tk -I$(HDIR) $(ODIR_DEBUG) 
+CFLAGS	=  -I../gcl-tk -I$(HDIR) $(ODIR_DEBUG) $(OPTFLAGS)
 
 
 MP 	= $(HDIR)mp.h
diff -urN gcl-2.4.0/unixport/makefile gcl-2.4.0-new/unixport/makefile
--- gcl-2.4.0/unixport/makefile	Mon Dec  3 22:53:05 2001
+++ gcl-2.4.0-new/unixport/makefile	Mon Dec  3 23:10:22 2001
@@ -7,7 +7,7 @@
 MPDIR	= ../mp
 GMP_DIR	= ../gmp/
 
-CFLAGS	= -c $(DEFS) -I$(HDIR)
+CFLAGS	= -c $(DEFS) -I$(HDIR) $(OPTFLAGS)
 LDCC    = $(CC) -g
 LIBC	= -lc	
 LIBS	= -lm # -lpixrect -lc
