--- gcl-2.4.3/configure.in.orig	Sat Jun  8 17:48:56 2002
+++ gcl-2.4.3/configure.in	Sun Jun 23 00:46:07 2002
@@ -158,7 +158,7 @@
 #--------------------------------------------------------------------
 
 AC_CHECK_FUNC(BSDgettimeofday, AC_DEFINE(HAVE_BSDGETTIMEOFDAY),
-	AC_CHECK_FUNC(gettimeofday, , AC_DEFINE(NO_GETTOD)))
+	[AC_CHECK_FUNC(gettimeofday, , AC_DEFINE(NO_GETTOD))])
 AC_MSG_CHECKING([for gettimeofday declaration])
 AC_EGREP_HEADER(gettimeofday, sys/time.h, AC_MSG_RESULT(present), [
     AC_MSG_RESULT(missing)
@@ -879,20 +879,14 @@
 
 MP_INLCUDE=""
 if test $use_gmp = yes ; then
- AC_MSG_CHECKING([use_gmp=yes, doing configure in gmp directory])
- case "${canonical}" in
-#   i[[5-9]]86* | pentium* | k6* | athlon*)
-#      (cd gmp ; ./configure --target=i486) ;;
-   *)
-      (cd gmp ; ./configure) ;;
- esac
- [[ "`ls -1 gmp/mpn/add_n.* 2>/dev/null`" != "" ]] || cp gmp/mpn/generic/*.c gmp/mpn/
+ AC_MSG_CHECKING("use_gmp=yes, checking for libgmp")
+ AC_CHECK_LIB(gmp,mpz_gcd,[],true)
  GMP=1
  AC_DEFINE(GMP)
  AC_SUBST(GMP)
  MP_INCLUDE=h/gmp.h
  echo > makedefsafter
- echo 'MPFILES=${GMP_DIR}libgmp.a' >> makedefsafter
+ echo 'MPFILES=-lgmp' >> makedefsafter
  echo >> makedefsafter
 fi
 AC_SUBST(MP_INCLUDE)
