diff -urN gcl-2.4.0/makefile gcl-2.4.0.new/makefile
--- gcl-2.4.0/makefile	Sun May  6 08:18:49 2001
+++ gcl-2.4.0.new/makefile	Sat Nov 17 10:50:38 2001
@@ -133,10 +133,10 @@
 	(cd $(BINDIR); make all)
 	make mpfiles
 	rm -f o/cmpinclude.h ; ln h/cmpinclude.h o
-	(cd $(ODIR); make all)
-	(cd $(LSPDIR); make all)
-	(cd $(CMPDIR); make all)
-	(cd $(PORTDIR); make saved_gcl)
+	make -C $(ODIR) all
+	make -C $(LSPDIR) all
+	make -C $(CMPDIR) all
+	make -C $(PORTDIR) saved_gcl
 	make command
 
 # 	After making successfully send in a notification to
