--- gcl/makefile.orig	2013-11-11 15:55:48.000000000 +0100
+++ gcl/makefile	2013-11-17 16:06:26.011391163 +0100
@@ -64,27 +64,27 @@
 #	cp $(<D)/../include/$(@F) $@
 
 $(PORTDIR)/saved_pre_gcl: $(HDIR)cmpinclude.h
-	(cd $(BINDIR); $(MAKE) all)
+	$(MAKE) -C $(BINDIR) all
 	$(MAKE) mpfiles
 	rm -f o/cmpinclude.h ; cp h/cmpinclude.h o
-	(cd $(ODIR); $(MAKE) all)
+	$(MAKE) -C $(ODIR) all
 	$(MAKE) $<
 	rm -f o/cmpinclude.h ; cp h/cmpinclude.h o
-	(cd $(ODIR); $(MAKE) all)
+	$(MAKE) -C $(ODIR) all
 	cd $(@D) && $(MAKE) $(@F)
 
 $(PORTDIR)/saved_gcl: $(PORTDIR)/saved_pre_gcl $(HDIR)cmpinclude.h 
-	(cd $(LSPDIR); touch *.lsp ; $(MAKE) all)
-	(cd $(CMPDIR); touch *.lsp ; $(MAKE) all)
-	[ -z "$(X_LIBS)" ] || (cd xgcl-2 && $(MAKE) LISP=../$<)
+	(cd $(LSPDIR); touch *.lsp) ; $(MAKE) -C $(LSPDIR) all
+	(cd $(CMPDIR); touch *.lsp) ; $(MAKE) -C $(CMPDIR) all
+	[ -z "$(X_LIBS)" ] || $(MAKE) -C xgcl-2 LISP=../$<
 	cd $(@D) && $(MAKE) $(@F)
 
 $(PORTDIR)/saved_pcl_gcl: $(PORTDIR)/saved_gcl
-	(cd $(PCLDIR); rm -f *.c; $(MAKE) all)
+	(cd $(PCLDIR); rm -f *.c) ; $(MAKE) -C $(PCLDIR) all
 	cd $(@D) && $(MAKE) $(@F)
 
 $(PORTDIR)/saved_ansi_gcl: $(PORTDIR)/saved_pcl_gcl
-	(cd $(CLCSDIR); rm -f *.c; $(MAKE) all)
+	(cd $(CLCSDIR); rm -f *.c) ; $(MAKE) -C $(CLCSDIR) all
 	cd $(@D) && $(MAKE) $(@F)
 
 ansi-tests/test_results: $(PORTDIR)/saved_ansi_gcl
@@ -116,7 +116,7 @@
 mpfiles: $(MPFILES)
 
 $(MPDIR)/libmport.a:
-	(cd mp ; $(MAKE) all)
+	$(MAKE) -C mp all
 
 $(GMPDIR)/libgmp.a: $(GMPDIR)/Makefile
 	cd $(GMPDIR) && $(MAKE) && rm -f libgmp.a &&  ar qc libgmp.a *.o */*.o
@@ -253,7 +253,7 @@
 OTHERS=$(HDIR)notcomp.h $(HDIR)rgbc.h $(HDIR)stacks.h 
 
 $(HDIR)new_decl.h:
-	(cd o && $(MAKE) ../$@)
+	$(MAKE) -C o ../$@
 
 $(HDIR)cmpinclude.h: $(CMPINCLUDE_FILES) $(HDIR)config.h
 	cat $(HDIR)config.h | sed -e "1,/Begin for cmpincl/d" -e "/End for cmpinclude/,50000d" > $(@F)
@@ -278,7 +278,7 @@
 	chmod a+rx configure
 
 kcp:
-	(cd go ; $(MAKE)  "CFLAGS = -I../h -pg  -c -g ")
-	(cd unixport ; $(MAKE) gcp)
+	$(MAKE) -C go "CFLAGS = -I../h -pg  -c -g "
+	$(MAKE) -C unixport gcp
 
 #.INTERMEDIATE: unixport/saved_pcl_gcl
